src/store/modules/routingSlip.ts

```diff
@@ -150,6 +150,32 @@ export default class RoutingSlipModule extends VuexModule {
     return this.routingSlip?.status === SlipStatus.VOID
   }
 
+  @Mutation
+  public updateRoutingSlipChequeNumber (chequeNumToChange) {
+    const payments = (this.routingSlip as RoutingSlip)?.payments.map((payment: Payment, i: number) => {
+      if (chequeNumToChange.idx === i) {
+        payment.chequeReceiptNumber = chequeNumToChange.chequeNum
+      }
+      return { ...payment }
+    })
+    this.routingSlip.payments = payments
+  }
+
+  @Mutation
+  public updateRoutingSlipAmount (amountToChange) {
+    const payments = (this.routingSlip as RoutingSlip)?.payments.map((payment: Payment, i: number) => {
```

Comment:
any reason why we need to cast this.routingSlip to RoutingSlip? Shouldn't it be defined as a routing slip already? 

This logic seems a bit weird.. how are you distinguishing if it's the USD or the CAD field?

Reply:
yeah it's defined already, so I just removed the cast;

updated the AdjustRoutingSlipAmountPrams interface, added isPaidinUsd attribute. So I can removed the logic of distinguishing if it's the USD or the CAD field.